addons/stock/report/report_lot_barcode.xml

```diff
@@ -19,15 +19,29 @@
                                 <t t-set="o" t-value="page_docs[0]"/>
                             </t>
                             <td t-att-style="barcode_index &gt;= len(page_docs) and 'visibility:hidden'">
-                                <div t-att-style="'position:relative; width:43mm; height:34mm; border: 1px solid %s;' % (o.env.user.company_id.primary_color or 'black')">
+                                <div t-att-style="'position:relative; width:43mm; height:19mm; border: 1px solid %s;' % (o.env.user.company_id.primary_color or 'black')">
                                     <t t-set="final_barcode" t-value="''"/>
                                     <t t-if="env.user.has_group('stock.group_stock_lot_print_gs1')">
                                         <t t-if="o.product_id.valid_ean" t-set="final_barcode" t-value="'01' + '0' * (14 - len(o.product_id.barcode)) + o.product_id.barcode"/>
                                         <t name="gs1_datamatrix_lot" t-if="o.product_id.tracking == 'lot'" t-set="final_barcode" t-value="(final_barcode or '') + '10' + o.name"/>
                                         <t t-elif="o.product_id.tracking == 'serial'" t-set="final_barcode" t-value="(final_barcode or '') + '21' + o.name"/>
                                     </t>
-                                    <span class="o_label_4x12" t-field="o.product_id.display_name" t-att-style="'width:22mm' if final_barcode else ''">Demo Product</span>
-                                    <span class="o_label_4x12" name="lot_name" t-field="o.name" t-att-style="'width:22mm' if final_barcode else ''">Demo Name</span>
+                                    <div class="o_label_4x12">
+                                        <t t-if="env.user.has_group('stock.group_stock_lot_print_gs1')">
+                                            <t t-set="code" t-value="o.product_id.default_code"/>
+                                            <t t-set="product_name" t-value="o.product_id.name"/>
```

Comment:
I think this template is already complicated enough without extra variables to keep track of. Please change the logic of this back to the way it was before the change that removed it https://github.com/odoo/odoo/pull/135739/commits/77ed573bd26cf79d3bce919ab1ad45a9baa9a885#diff-fbc7f34c633189f45e17080184e52bb403bf631e36d8713cb0b2d27260629a3b

For reference, this is how the name was originally split: https://github.com/odoo/odoo/pull/122511/commits/9ffb7110a8b0f5f8aeb6418fd2c73338a69c7068